Foundation/NSString.swift

```diff
@@ -1237,9 +1237,7 @@ extension NSString {
     }
     
     public convenience init?(CString nullTerminatedCString: UnsafePointer<Int8>, encoding: UInt) {
-        guard let cf = CFStringCreateWithCString(kCFAllocatorSystemDefault, nullTerminatedCString, CFStringConvertNSStringEncodingToEncoding(encoding)) else {
-            return nil
-        }
+        let cf = CFStringCreateWithCString(kCFAllocatorSystemDefault, nullTerminatedCString, CFStringConvertNSStringEncodingToEncoding(encoding))
```

Comment:
CFStringCreateWithCString can and will return NULL when it cannot convert the cString to the requested encoding. So the call later to cf._nsObject will fail